esp/services/WsDeploy/WsDeployEngine.cpp

```diff
@@ -206,8 +206,7 @@ void CWsDeployEngine::deploy(CDeployOptions& pOptions)
 {
   try
   {
-    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
-    factory->validateCache();   
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
     Owned<IConstEnvironment> constEnv(factory->openEnvironment());
```

Comment:
See comment below.  Following code is invalid if it returns null

Reply:
agree, it should be assume must return non null, if openEnvironment could fail, it should throw an exception.